pkg/querier/distributor_queryable.go

```diff
@@ -86,7 +86,10 @@ func (q *distributorQuerier) Select(_ bool, sp *storage.SelectHints, matchers ..
 	spanlog, ctx := spanlogger.NewWithLogger(q.ctx, q.logger, "distributorQuerier.Select")
 	defer spanlog.Finish()
 
-	minT, maxT := sp.Start, sp.End
+	minT, maxT := q.mint, q.maxt
+	if sp != nil {
```

Comment:
If `querier.Select` replaces `nil` sp with non-nil one, can `sp` be still `nil` here? (Just curious why is this change required)

Reply:
needed purely for consistency. So that `distributorQuerier` is an independent implementation of `Querier` that works ~on its own. I may be talking nonsense in case the two are super coupled. I have not dived into the code

Reply:
They look independant, but they are always used together right now. I don't mind the change, I was just curious if it is needed. Feel free to keep it.

Reply:
I also prefer to keep it, as Dimitar suggests.